disk-buffering/src/main/java/io/opentelemetry/contrib/disk/buffering/internal/storage/exceptions/MaxAttemptsReachedException.java

```diff
@@ -0,0 +1,11 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.contrib.disk.buffering.internal.storage.exceptions;
+
+import java.io.IOException;
+
+@SuppressWarnings("serial")
+public class MaxAttemptsReachedException extends IOException {}
```

Comment:
using exception for control flow is discouraged in Java. Can we use some kind of result enum or object instead?

Reply:
I wonder if this applies for any kind of flow? I'm not aware of the details of the discouragement for this cases, but the reason I decided to use exceptions for non-happy path cases is because it allows for a more straightforward API usage. For example, this exception is thrown [in here](https://github.com/open-telemetry/opentelemetry-java-contrib/pull/913/files/549ea97609ea7ee71ae2e069fd1e8cfb4ea2c9da#diff-4ab92b18edc8e550a570fd5e6f577e0d5d08df1f1aed47fd700ffe3c79e0694bR45) in case that some unexpected amount of recursive `write` calls happen (which in theory shouldn't, but it's there to make it more robust). Though essentially, `write` should not have to return anything because its happy path would just complete without any exceptions. If we make it to return an enum for example, some manual verification has to be done on the caller side in order to avoid ignoring an unexpected scenario, and the compiler won't be able to help by forcing the caller to handle that scenario unlike what happens with this exception. So generally speaking I'm always in favor to raise issues when something unexpected happens to make sure they are easily visible and taken into account at the time of writing the code, seems to make for a safer runtime.

Reply:
here's an article: https://softwareengineering.stackexchange.com/questions/189222/are-exceptions-as-control-flow-considered-a-serious-antipattern-if-so-why

In my experience, it's much harder to understand control flow with exceptions, because an exception can terminate multiple layers in the call stack.
On the other hand, it's very easy to check that a returned result code is not ignored.

Reply:
Thank you, I've made the changes.